src/StreamMaster.cpp

```diff
@@ -0,0 +1,203 @@
+#include "StreamMaster.h"
+#include "Streamer.h"
+
+FileWriter::StreamMaster::StreamMaster(
+    const std::string &Broker, std::unique_ptr<FileWriterTask> FileWriterTask,
+    const MainOpt &Options,
+    std::unique_ptr<StreamerBaseFactory> StreamerFactory)
+    : Demuxers(FileWriterTask->demuxers()),
+      ProducerTopic(FileWriterTask->getStatusProducer()),
+      WriterTask(std::move(FileWriterTask)),
+      TopicWriteDuration{Options.topic_write_duration},
+      ServiceId{Options.service_id} {
+
+  for (auto &Demux : Demuxers) {
+    try {
+      Streamers.emplace(std::piecewise_construct,
```

Comment:
Why do we have to use `std::piecewise_construct` and `std::forward_as_tuple` if the member is declared as `std::map<std::string, std::unique_ptr<StreamerBase>> Streamers` and `StreamerFactory` already returns `std::unique_ptr<FileWriter::StreamerBase>`?